src/utils/polling.js

```diff
@@ -36,45 +72,56 @@ async function pollSubscriptionTopics() {
     (s) => s.extension && s.extension.some((e) => e.url === TOPIC_URL)
   );
 
-  // Remove duplicates with Set so we don't poll twice for same topic
-  const topicsToPoll = [
+  // Map topics to resources and remove duplicates with Set so we only poll once for each resource
+  const resourcesToPoll = [
     ...new Set(
-      subscriptions.map((s) => {
-        const topicExtension = s.extension.find((e) => e.url === TOPIC_URL);
-
-        return topiclist.parameter.find(p => p.valueCanonical === topicExtension.valueUri).name;
-      })
+      subscriptions
+        .map((s) => {
+          const topicExtension = s.extension.find((e) => e.url === TOPIC_URL);
+          return topiclist.parameter.find((p) => p.valueCanonical === topicExtension.valueUri).name;
+        })
+        .map((t) => namedEventToResourceType(t))
     ),
   ];
 
-  if (topicsToPoll.length === 0) {
+  if (resourcesToPoll.length === 0) {
     logger.info('No subscription topics to poll.');
     return;
   }
 
   const { baseUrl, clientId } = fhirClientConfig;
   const accessToken = await getAccessToken(baseUrl, clientId);
-  topicsToPoll.forEach((topic) => {
-    logger.info(`Polling EHR for ${topic}.`);
+  resourcesToPoll.forEach((resourceToPoll) => {
+    logger.info(`Polling EHR for ${resourceToPoll} resources.`);
```

Comment:
Last thing which is super minor - can this line be moved a little further down so it can include the `lastUpdated` parameter. Would be helpful to see in the log exactly what is being queried